aptos-move/framework/aptos-names/sources/domains.move

```diff
@@ -148,13 +153,27 @@ module aptos_names::domains {
         assert!(is_valid, error::invalid_argument(EDOMAIN_HAS_INVALID_CHARACTERS));
         assert!(length <= config::max_domain_length(), error::out_of_range(EDOMAIN_TOO_LONG));
 
+        let account_address = signer::address_of(sign);
+        // if unrestricted_mint_enabled() is false, or if we are calling this function
+        // from `register_domain_with_signature` with a `signature` of length 64,
+        // verify the signature
+        if (!unrestricted_mint_enabled() || vector::length(&signature) == 64) {
```

Comment:
Actually, why even check the signature length? This will not work with other signature schemes, should we add them later....

Reply:
was thinking that the public key will be ed25519, so the signature will be length of 64... checking so that if there's an non-empty signature we'll verify it, but I will update it to check `vector:length(&signature) != 0`

Reply:
I've removed all the length checks! yeah I was more thinking about saving gas, but I guess it isn't really worth it in this case